server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -389,7 +397,8 @@ private boolean canRemoveTombstone(long maxTimestampToPrune, long maxSeqNoToPrun
      * Try to prune tombstones whose timestamp is less than maxTimestampToPrune and seqno at most the maxSeqNoToPrune.
      */
     void pruneTombstones(long maxTimestampToPrune, long maxSeqNoToPrune) {
-        for (Map.Entry<BytesRef, DeleteVersionValue> entry : tombstones.entrySet()) {
+        final Tombstones tombstones = maps.tombstones;
```

Comment:
Similarly - this feels like it could move onto `Maps` or even `Tombstones` - AFAICT the only other thing it needs is `maps.getMinDeleteTimestamp()`.